extensions/amp-pan-zoom/0.1/amp-pan-zoom.js

```diff
@@ -644,39 +662,38 @@ export class AmpPanZoom extends AMP.BaseElement {
   }
 
   /**
-   * Performs a one-step pinch zoom action.
-   * @param {number} centerClientX
-   * @param {number} centerClientY
-   * @param {number} deltaX
-   * @param {number} deltaY
-   *  @param {number} dir
-   * @private
+   * @param {number} clientX
+   * @param {number} clientY
    */
-  onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir) {
-    this.zoomToPoint_(centerClientX, centerClientY, deltaX, deltaY, dir);
+  onDoubletapZoom_(clientX, clientY) {
+    const newScale = this.scale_ == 1 ? this.maxScale_ : this.minScale_;
+    const dx = (this.elementBox_.width / 2) + this.getOffsetX_(clientX);
+    const dy = (this.elementBox_.height / 2) + this.getOffsetY_(clientY);
+    return this.onZoom_(newScale, dx, dy, /*animate*/ true);
   }
 
   /**
-   * Given center position, zoom delta, and zoom position, computes
-   * and updates a zoom action on the content.
+   * Performs a one-step pinch zoom action.
    * @param {number} centerClientX
    * @param {number} centerClientY
    * @param {number} deltaX
    * @param {number} deltaY
-   * @param {number} dir
+   *  @param {number} dir
    * @private
    */
-  zoomToPoint_(centerClientX, centerClientY, deltaX, deltaY, dir) {
+  onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir) {
     if (dir == 0) {
       return;
     }
+    const {width, height} = this.elementBox_;
     const dist = Math.sqrt((deltaX * deltaX) + (deltaY * deltaY));
-    const newScale = this.startScale_ * (1 + (dir * dist / 100));
-    const deltaCenterX = (this.elementBox_.width / 2) - centerClientX;
-    const deltaCenterY = (this.elementBox_.height / 2) - centerClientY;
-    deltaX = Math.min(deltaCenterX, deltaCenterX * (dist / 100));
-    deltaY = Math.min(deltaCenterY, deltaCenterY * (dist / 100));
-    this.onZoom_(newScale, deltaX, deltaY, /*animate*/ false);
+    const pinchFactor = dist / 100;
```

Comment:
Not really sure what pinch factor means. What situations lead it to be greater than 1 (which is capped below for dx, dy)? Why the constant 100?

Reply:
100 is an arbitrary tweaked based on UX heuristics for pinch zoom. It determines how fast the zoom happens relative to the pinch. I guess I could use a better name for it.

Reply:
By default, I want to zoom in such a way as to treat the center of my pinch as the new center. However, I do not want to move by too much. So how much I move is dampened by how much I actually pinched. We never want to move past my new pinch center, hence the pinch factor here is capped at 1. The point of applying the pinch factor here is to influence how much the content moves by how much we pinched, and the 100 is more like an empirical UX heuristic. 

I've reverted this back to `dist / 100` now. I could pull `100` out into a constant.